### PR TITLE
values in the `sample_link` slot may not always follow a formal syntax, like CURIe or GUID

### PR DESCRIPTION
**Pull request by bmeluch**
From 
- https://github.com/microbiomedata/nmdc-schema/issues/1669

I updated the slot definition as described in the original issue, then updated the valid example files that already contained `sample_link` and created an invalid example file.

### CODE REVIEW

**Review by turbomam — COMMENTED** [review]
Thanks @bmeluch 

I expect that I can merge this in very soon, but I would like to touch base with you and any other stakeholders first.

Your examples illustrate three different kinds of relationships (as the prefix portion of the values), but is that an exhaustive list? Is there some documentation or code snippet that shows how the different kinds of relationships will be expressed in data? Where will we find the part of the sample linkage value that comes after the colon? Like, **which** slot of some other Biosample I guess?

**Comment by bmeluch** [comment]
Hi @turbomam, 
Good question - I would have to double check with @mslarae13 about the complete list of relationships (before colon). I believe the part after the colon has to match `samp_name` (s) of the linked biosamples.

File: src/data/valid/Biosample-exhasutive.yaml [hunk 1]
```diff
@@ -876,8 +875,7 @@ samp_well_name:
   has_raw_value: xxx
 
 sample_link:
-  - IGSN:DSJ0284
-  - any:curie
+  - 'core: soil_core_O'
```
Comment (mslarae13) [inline src/data/valid/Biosample-exhasutive.yaml:878]: I would expect core to be source_mat_id. so i think this one should be removed. Maybe replace core with plant:

File: src/data/valid/Biosample-exhaustive-issue-796-bye-yq-for-7-4-10.yaml [hunk 2]
```diff
@@ -380,9 +379,9 @@ technical_reps: "2"
 analysis_type:
   - metabolomics
   - metagenomics
+
 sample_link:
-  - IGSN:DSJ0284
-  - any:curie
+  - 'parent: example_samp_45'
```
Comment (mslarae13) [inline src/data/valid/Biosample-exhaustive-issue-796-bye-yq-for-7-4-10.yaml:384]: parent is source_mat_id ... 1:1 parent to child. 
sample link is other sample types or horizontal relationships.. maybe make this one sibling:

File: src/schema/portal/sample_id.yaml [hunk 3]
```diff
@@ -46,16 +46,19 @@ slots:
...
     comments:
       - 'This field allows multiple entries separated by ; (Examples: Soil collected
       from the field will link with the soil used in an incubation. The soil a plant
       was grown in links to the plant sample. An original culture sample was transferred
       to a new vial and generated a new sample)'
     examples:
-      - value: IGSN:DSJ0284
+      - value: 'core: soil core 1'
+      - value: 'parent: sample A'
+      - value: 'pooling: sample 1, sample 2, sample 3'
```
Comment (mslarae13) [inline src/schema/portal/sample_id.yaml:61]: pooling is good!

**Review by mslarae13 — CHANGES_REQUESTED** [review]

**Comment by mslarae13** [comment]
@bmeluch ignore my comments until we chat. I need to review the NMDC schema and submission portal. source_mat_id needs some work. We can discuss this this week but I think we'll make a squad to do this.

**Comment by turbomam** [comment]
> We can discuss this this week but I think we'll make a squad to do this.

Could I be part of this conversation?

File: src/data/invalid/Biosample-invalid-linkage.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    id: ENVO:00002030
+env_local_scale:
+  has_raw_value: ENVO:00002169
+  term:
+    id: ENVO:00002169
+env_medium:
+  has_raw_value: ENVO:00005792
+  term:
+    id: ENVO:00005792
+sample_link:
```
Comment (turbomam) [inline src/data/invalid/Biosample-invalid-linkage.yaml:16]: is this a list of three free stating strings and an integer?

**Review by turbomam — COMMENTED** [review]
I asked about the vocabulary of relationship types before, but now I think we need to discuss the overall grammar, and what agent (person or software) is responsible for interpreting it.

It wouldn't be too hard to do that with structured objects in the nmdc-schema, but it  will be harder in the flattened submission-schema. Are you planning on allowing different relationships to be semi-colon delimited in the DH cells? Will there be any entry-time validation?

**Comment by turbomam** [comment]
We defined `sample_link` in `src/schema/portal/sample_id.yaml`, so we can do whatever we want with it, right? Let's start making a distinction between a more structured modeling in nmdc-schema and move this flattened approach to the submission-schema, where it will be totally relevant and appropriate.

**Comment by bmeluch** [comment]
This is now way out of date, and we need more discussion on `sample_link` - closing this but will leave the issue open and start a new PR when the discussion comes back around.

**Pull request closed without merging**